File: weyl-core/src/main/java/io/quartic/weyl/core/compute/BucketSpec.java
package io.quartic.weyl.core.compute;

import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
import com.fasterxml.jackson.databind.annotation.JsonSerialize;
import io.quartic.common.SweetStyle;
import io.quartic.weyl.core.model.LayerId;
import org.immutables.value.Value;

@SweetStyle
@Value.Immutable
@JsonSerialize(as = BucketSpecImpl.class)
@JsonDeserialize(as = BucketSpecImpl.class)
public interface BucketSpec extends ComputationSpec {
    LayerId buckets();
    LayerId features();
    BucketAggregation aggregation();
    boolean normalizeToArea();
}

File: weyl-core/src/main/java/io/quartic/weyl/core/compute/BucketSpec.kt
package io.quartic.weyl.core.compute

import io.quartic.weyl.core.model.LayerId

data class BucketSpec (
        val buckets: LayerId,
        val features: LayerId,
        val aggregation: BucketAggregation,
        val normalizeToArea: Boolean
) : ComputationSpec
